src/App.tsx

```diff
@@ -21,55 +26,143 @@ enum SortType {
   LENGTH,
 }
 
-// Use this function in the render method
+type State = {
+  isStarted: boolean,
+  isReversed: boolean,
+  sortType: SortType,
+};
+
 function getReorderedGoods(
   goods: string[],
   sortType: SortType,
   isReversed: boolean,
 ) {
-  // Not to mutate the original array
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // ...
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPABET:
+        return good1.localeCompare(good2);
+
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      case SortType.NONE:
+      default:
+        return 0;
+    }
+  });
```

Comment:
Your sort will start every time and will check each element and do nothing
```suggestion
    switch (sortType) {
      case SortType.ALPABET:
        return visibleGoods.sort((a, b) => a.localeCompare(b));
      case SortType.LENGTH:
        return visibleGoods.sort((a, b) => a.length - b.length);
      case SortType.NONE:
      default:
        break;
    }
```

Reply:
working this way:
  switch (sortType) {
    case SortType.ALPABET:
      visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
      **break;**
    case SortType.LENGTH:
      visibleGoods.sort((good1, good2) => good1.length - good2.length);
      **break;**
    case SortType.NONE:
    default:
      break;
  }